lib/logstash/inputs/s3.rb

```diff
@@ -111,7 +111,7 @@ def list_new_files
       @logger.debug("S3 input: Found key", :key => log.key)
 
       unless ignore_filename?(log.key)
-        if sincedb.newer?(log.last_modified)
+        if sincedb.newer?(log.last_modified) && log.content_size > 0
```

Comment:
If I read the documentation carefully it should be `content_length`
http://docs.aws.amazon.com/AWSRubySDK/latest/AWS/S3/S3Object.html#content_length-instance_method